modules/processor/src/main/java/uk/gov/hmcts/reform/migration/CaseMigrationProcessor.java

```diff
@@ -36,30 +48,53 @@ public void processSingleCase(String userToken, String caseId) {
-            .forEach(caseDetails -> updateCase(userToken, caseDetails.getId(), caseDetails.getData()));
+    public void processAllCases(String userToken, String userId, String caseType) {
+        coreCaseDataService.fetchAll(userToken, userId, caseType).stream()
+            .forEach(caseDetails -> updateCase(userToken, caseType, caseDetails));
     }
 
-    private void updateCase(String authorisation, Long id, Map<String, Object> data) {
-        log.info("Updating case {}", id);
-        try {
-            log.debug("Case data: {}", data);
-            coreCaseDataService.update(authorisation, id.toString(),
-                EVENT_ID, EVENT_SUMMARY, EVENT_DESCRIPTION, dataMigrationService.migrate(data));
-            log.info("Case {} successfully updated", id);
-            migratedCases.add(id);
-        } catch (Exception e) {
-            log.error("Case {} update failed due to: {}", id, e.getMessage());
-            failedCases.add(id);
+    public void migrateCases(String caseType) {
+        log.info("Data migration of all cases started for case type: {}", caseType);
+        String userToken =  idamRepository.generateUserToken();
+        List<CaseDetails> listOfCaseDetails = elasticSearchRepository.findCaseByCaseType(userToken, caseType);
+        listOfCaseDetails.stream()
+            .forEach(caseDetails -> updateCase(userToken, caseType, caseDetails));
+        log.info(LOG_STRING);
+        log.info("Data migration completed");
+        log.info(LOG_STRING);
+        log.info("Total number of processed cases: {}", getMigratedCases().size() + getFailedCases().size());
+        log.info("Total number of migrations performed: {}", getMigratedCases().size());
+        log.info(LOG_STRING);
+        log.info("Migrated cases: {} ", !getMigratedCases().isEmpty() ? getMigratedCases() : "NONE");
+        log.info("Failed cases: {}", !getFailedCases().isEmpty() ? getFailedCases() : "NONE");
+        log.info("Data migration of all cases completed");
+    }
+
+    private void updateCase(String authorisation, String caseType, CaseDetails caseDetails) {
+        if (dataMigrationService.accepts().test(caseDetails)) {
+            Long id = caseDetails.getId();
+            log.info("Updating case {}", id);
+            try {
+                log.debug("Case data: {}", caseDetails.getData());
+                coreCaseDataService.update(
+                    authorisation,
+                    EVENT_ID,
+                    EVENT_SUMMARY,
+                    EVENT_DESCRIPTION,
+                    caseType,
+                    caseDetails
+                );
+                log.info("Case {} successfully updated", id);
+                migratedCases.add(id);
+            } catch (Exception e) {
+                log.error("Case {} update failed due to: {}", id, e.getMessage());
+                failedCases.add(id);
+            }
+        } else {
+            log.info("Case {} already migrated", caseDetails.getId());
```

Comment:
Minor - because it doesn't necessarily mean that the case is already migrated, it could also be that the service implemented some rules that make it not applicable for other reasons.
```suggestion
            log.info("Case {} does not meet criteria for migration", caseDetails.getId());
```

Reply:
done